lib/yi_appium_caps_util.rb

```diff
@@ -122,9 +134,29 @@ def update_ios_caps (output_data)
             ipcounter = 999
           end
         end
-        File.delete('iplog.txt')
-        #Replace value of udid
-        output_data['caps']['youiEngineAppAddress'] = ip
+        # Get the device name before deleting the file
+        deviceName = %x[grep -Eo "deviceName =.*" logs.txt| head -1|cut -d " " -f 3-|tr -d '"'|tr -d '\n']
+        platformVersion = %x[grep -Eo "platformVersion =.*" logs.txt| head -1|awk '{print $3}'| tr -d '"'|tr -d '\n']
+        puts 'DeviceName: ' + deviceName
+        puts 'platformVersion: ' + platformVersion
+        File.delete('logs.txt')
+        #Replace value of ip if found
+        if (ip != '') then 
+          output_data['caps']['youiEngineAppAddress'] = ip 
+        else 
+          puts "Update ip address manually"
+        end
+        output_data['caps']['deviceName'] = deviceName
+        output_data['caps']['platformVersion'] = platformVersion
+
+        # Add the xcodeConfigFile in the caps if dealing with iOS 10+
+        if (platformVersion.to_f>=10) then
+          output_data['caps']['xcodeConfigFile'] = './youi.xcconfig'
+          # NewWDA: Forces uninstall of any existing WebDriverAgent app on device. This provides stability.
+          output_data['caps']['useNewWDA'] = true
+          temp = "DEVELOPMENT_TEAM = W4E9HL2DXS\nCODE_SIGN_IDENTITY = iPhone Developer\n"
+          File.write('youi.xcconfig', temp)
+        end
```

Comment:
@simongranger Do you think it will add value if have an else (for <10) and output a reminder to the user to switch to xcode 7?